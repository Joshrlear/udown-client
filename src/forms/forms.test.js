import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount, render } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Login from './login';
import Signup from './signup';

configure({ adapter: new Adapter() })

describe('Components', () => {
    const mappedComponents = ['Login', 'Signup']
    mappedComponents.map(Component => {
        it(`renders Components without crashing`, () => {
            const div = document.createElement('div')
            ReactDOM.render(
            <BrowserRouter>
               <Component/>
            </BrowserRouter>, div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })

        it(`handleInput should update state.username`, () => {
            const wrapper = mount(<Login />, {
                disableLifecycleMethods: true,
            })
            .instance();
            const input = wrapper.find('.username');
            expect(input.simulate('change', {
                target: { name: "username", value: "josh" }
            }))
            console.log(wrapper.state)
            expect(wrapper.state.username).toBe({
                'username': 'josh'
            })
        })

        it(`handleInput should update state.username`, () => {
            const wrapper = shallow(<Signup />, {
                disableLifecycleMethods: true,
            })
            .instance();
            wrapper.handleInput({target: { name: "username", value: "josh" }})
            console.log(wrapper.state)
            expect(wrapper.state.username).toBe({
                'username': 'josh'
            })
        })
})