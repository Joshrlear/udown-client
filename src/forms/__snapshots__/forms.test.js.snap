// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components renders Login correctly 1`] = `
<div
  className="login_container"
>
  <section
    className="test_account_info"
  >
    <h3>
      Test account
    </h3>
    <p>
      username: test
    </p>
    <p>
      password: password
    </p>
  </section>
  <form
    className="login_form"
    onSubmit={[Function]}
  >
    <h2>
      Login
    </h2>
    <ErrorMsg />
    <div>
      <label>
        Username: 
      </label>
      <input
        className="username"
        name="username"
        onChange={[Function]}
        placeholder="username here"
      />
    </div>
    <br />
    <div>
      <label>
        Password: 
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="password here"
        type="password"
      />
    </div>
    <br />
    <div
      className="button_container"
    >
      <button
        type="submit"
      >
        Login
      </button>
    </div>
    <br />
    <NavLink
      className="link"
      to="/signup"
    >
      Go to SignUp
    </NavLink>
  </form>
</div>
`;

exports[`Components renders Signup correctly 1`] = `
<div
  className="signup_container"
>
  <form
    className="signup_form"
    onSubmit={[Function]}
  >
    <h2>
      Signup
    </h2>
    <ErrorMsg />
    <div>
      <label>
        Username: 
      </label>
      <input
        className="username"
        name="username"
        onChange={[Function]}
        placeholder="username here"
      />
    </div>
    <br />
    <div>
      <label>
        Password: 
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="password here"
        type="password"
      />
    </div>
    <br />
    <div>
      <label>
        Retype password: 
      </label>
      <input
        name="retype_password"
        onChange={[Function]}
        placeholder="password here"
        type="password"
      />
    </div>
    <br />
    <div
      className="button_container"
    >
      <button
        type="submit"
      >
        SignUp
      </button>
    </div>
    <br />
    <NavLink
      className="link"
      to="/login"
    >
      Go to Login
    </NavLink>
  </form>
</div>
`;
